lib/utils/device/live.ts

```diff
@@ -162,11 +164,31 @@ export class LivepushManager {
 					eventQueue.push(changedPath);
 					this.getDebouncedEventHandler(serviceName)();
 				};
+
+				// Prepare dockerignore data for file watcher
+				const serviceDirsByService = this.deployOpts.multiDockerignore
+					? await getServiceDirsFromComposition(
+							this.deployOpts.source,
+							this.composition,
+					  )
+					: {};
+				const { ignoreByDir, serviceDirs } = await getDockerignoreByDir(
+					this.deployOpts.source,
+					serviceDirsByService,
+				);
+
 				// TODO: Memoize this for containers which share a context
 				const monitor = chokidar.watch('.', {
 					cwd: context,
 					ignoreInitial: true,
-					ignored: '.git',
```

Comment:
Should we keep `.git` hard-coded to be ignored? What about adding others, such as `node_modules`, `.balena`, etc?

Reply:
The `getDockerIgnoreInstance` function already hardcodes `.git` and other patterns:
https://github.com/balena-io/balena-cli/blob/v12.27.2/lib/utils/ignore.ts#L264-L274

So when using dockerignore in the implementation of the chokidar `ignored` callback, those hardcoded patterns are automatically used. Actually, for the better or the worse (probably the worse), the following files or folders will not be ignored if present under your `node_modules` folders... :-/
```
		'!**/.balena',
		'!**/.resin',
		'!**/Dockerfile',
		'!**/Dockerfile.*',
		'!**/docker-compose.yml',
```